### PR TITLE
Develop refactor

### PR DESCRIPTION
**Pull request by dmigwi**

### CODE REVIEW

File: amity.py [hunk 1]
```diff
@@ -0,0 +1,339 @@
...
+
+
+class Amity(object):
+
+    def __init__(self):
+        self.engine = ''
+        self.felo = Fellow()
+        self.staf = Staff()
+        self.livin = LivingSpace()
+        self.offy = Office()
```
Comment (njirap) [inline amity.py:69]: have better variable names..
`self.fellow, self.staff, self.living_space and self.office`

File: amity.py [hunk 2]
```diff
@@ -0,0 +1,339 @@
...
+
+    def connect_db(self, db_name=None):
+        if db_name is not None:
+            name = 'sqlite:///' + db_name[:db_name.index('.')] + '.db'
+        else:
+            name = 'sqlite:///amity.db'
```
Comment (njirap) [inline amity.py:75]: `db_name = db_name[:db_name.index('.')] or 'amity'`
`return create_engine(db_name+'.db')`

File: amity.py [hunk 3]
```diff
@@ -0,0 +1,339 @@
...
+        engine = self.connect_db()
+
+        Base.metadata.create_all(engine)
+
+    def load_state(self):
+        '''copy data from the database to the application'''
+        engine = self.connect_db()
+
+        print("Please Wait... This may take some few minutes.")
+        # do something
```
Comment (njirap) [inline amity.py:88]: for example what?

File: amity.py [hunk 4]
```diff
@@ -0,0 +1,339 @@
...
+    def retrieve_data_from_db(self, session):
+        fellows = []
+        staffs = []
+        query_room = session.query(Room).all()
+        query_user = session.query(User).all()
+        for user in query_user:
+            if user.type_person == 'fellow':
+                fellows.append(user.person_name)
+            else:
+                staffs.append(user.person_name)
```
Comment (njirap) [inline amity.py:111]: `fellows = [user.person_name for user in query_user if user.type_person == 'fellow']`
`staff = [user.person_name for user in query_user if user.type_person == 'staff']`

File: amity.py [hunk 5]
```diff
@@ -0,0 +1,339 @@
...
+                    off.append(room.occupant3)
+
+                if room.occupant4 != '':
+                    off.append(room.occupant4)
+
+                if room.occupant5 != '':
+                    off.append(room.occupant5)
+
+                if room.occupant6 != '':
+                    off.append(room.occupant6)
```
Comment (njirap) [inline amity.py:136]: `office_occupants = [room_occupant for room_occupant in [room.occupant1,...,room.occupant6] if room_occupant != '']`

File: amity.py [hunk 6]
```diff
@@ -0,0 +1,339 @@
...
+
+                if room.occupant2 != '':
+                    liv.append(room.occupant2)
+
+                if room.occupant3 != '':
+                    liv.append(room.occupant3)
+
+                if room.occupant4 != '':
+                    liv.append(room.occupant4)
+                    livingspaces[room.room_name] = liv
```
Comment (njirap) [inline amity.py:152]: this can also be list comprehended

File: amity.py [hunk 7]
```diff
@@ -0,0 +1,339 @@
...
+        # load data from db
+        print(self.felo.load_fellows(fellows))
+        print(self.staf.load_staffs(staffs))
+        print(self.livin.load_livingspaces(livingspaces))
+        print(self.offy.load_offices(offices))
```
Comment (njirap) [inline amity.py:157]: `text = '%s'%self.felo.load_fellows(fellows)`
`text = text + '\n%s'%self.staf.load_staffs(staffs)`
`..`
print(text)

File: amity.py [hunk 8]
```diff
@@ -0,0 +1,339 @@
...
+        print("\tFinalizing database population....")
+
+        session.commit()
+        session.flush()
+        print("Save State successfully completed.")
+
+    def order_data_ready_for_saving(self, session):
+        people = []
+        people.extend(list(Fellow.fellow_names))
+        people.extend(list(Staff.staff_names))
```
Comment (njirap) [inline amity.py:182]: `people = Fellow.fellow_names + Staff.staff_names`

File: amity.py [hunk 9]
```diff
@@ -0,0 +1,339 @@
...
+        ordered_users = []
+        for person in people:
+            user = User()
+            user.person_name
+            user.type_person
```
Comment (njirap) [inline amity.py:187]: what is happening here and in the previous line?

File: amity.py [hunk 10]
```diff
@@ -0,0 +1,339 @@
...
+            ordered_users.append(user)
+
+        session.add_all(ordered_users)
+
+        Total_rooms = {}
```
Comment (njirap) [inline amity.py:192]: variable names start with small letters..

File: amity.py [hunk 11]
```diff
@@ -0,0 +1,339 @@
...
+        Total_rooms.update(Office.office_n_occupants)
+        Total_rooms.update(LivingSpace.room_n_occupants)
+        ordered_rooms = []
+        for key, value in list(Total_rooms.items()):
```
Comment (njirap) [inline amity.py:196]: ??????

File: amity.py (hunk 2 above)
Comment (dmigwi) [reply amity.py:75]: `db_name = db_name[:db_name.index('.')] or 'amity'` is raising bugs when db_name is equal to None
image

**Pull request merged**